analyzers/src/SonarAnalyzer.CSharp/Rules/InfiniteRecursion_RoslynCfg.cs

```diff
@@ -0,0 +1,224 @@
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using SonarAnalyzer.CFG.Helpers;
+using SonarAnalyzer.CFG.Roslyn;
+using SonarAnalyzer.Extensions;
+using SonarAnalyzer.Helpers;
+using StyleCop.Analyzers.Lightup;
+using CfgAllPathValidator = SonarAnalyzer.CFG.Roslyn.CfgAllPathValidator;
+
+namespace SonarAnalyzer.Rules.CSharp
+{
+    public partial class InfiniteRecursion
+    {
+        private class InfiniteRecursion_RoslynCfg : IInfiniteRecursion
+        {
+            public void CheckForNoExitProperty(SyntaxNodeAnalysisContext c, PropertyDeclarationSyntax property, IPropertySymbol propertySymbol)
+            {
+                if (property.ExpressionBody?.Expression != null)
+                {
+                    var cfg = ControlFlowGraph.Create(property.ExpressionBody, c.SemanticModel);
+                    var walker = new CfgWalkerForProperty(
+                        new RecursionAnalysisContext<ControlFlowGraph>(cfg, propertySymbol, property.Identifier.GetLocation(), c),
+                        "property's recursion",
+                        isSetAccessor: false);
+                    walker.CheckPaths();
+                    // cannot meet goto s here, check for unreachable exit node is not needed here
+                    return;
+                }
+
+                var accessors = property.AccessorList?.Accessors.Where(a => a.HasBodyOrExpressionBody());
+                if (accessors != null)
+                {
+                    foreach (var accessor in accessors)
+                    {
+                        var cfg = ControlFlowGraph.Create(accessor, c.SemanticModel);
+                        var walker = new CfgWalkerForProperty(
+                            new RecursionAnalysisContext<ControlFlowGraph>(cfg, propertySymbol, accessor.Keyword.GetLocation(), c),
+                            "property accessor's recursion",
+                            isSetAccessor: accessor.Keyword.IsKind(SyntaxKind.SetKeyword));
+                        if (!walker.CheckPaths() && !cfg.ExitBlock.IsReachable)
+                        {
+                            c.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, accessor.Keyword.GetLocation(), "property accessor's recursion"));
+                        }
+                    }
+                }
+            }
+
+            public void CheckForNoExitMethod(SyntaxNodeAnalysisContext c, CSharpSyntaxNode body, SyntaxToken identifier, ISymbol symbol)
+            {
+                var operation = c.SemanticModel.GetOperation(body.Parent);
+                var wrapper = new IOperationWrapperSonar(operation);
+                while (wrapper.Parent != null)
+                {
+                    wrapper = new IOperationWrapperSonar(wrapper.Parent);
+                }
+
+                var cfg = ControlFlowGraph.Create(wrapper.Instance.Syntax, c.SemanticModel);
+                var enclosingKinds = new HashSet<SyntaxKind>
```

Comment:
This method covers only the case when node is local function, all the rest are not covered at all, since they are not point of this code. I am okay doing this when this will be really usable and testable. Or as a bonus in this sprint when we have time.